pydatastructs/miscellaneous_data_structures/stack.py

```diff
@@ -109,3 +113,61 @@ def __str__(self):
         Used for printing.
         """
         return str(self.items._data)
+
+
+class LinkedListStack(Stack):
+
+    def __new__(cls, items=None, dtype=NoneType):
+        obj = object.__new__(cls)
+        obj.stack = SinglyLinkedList()
+        obj._dtype = dtype
+        obj.front = obj.stack.head
+        obj.tail = obj.stack.tail
```

Comment:
IMO, only `obj.top` should suffice and that too as a `@property` and not as a separate attribute.

Reply:
Agreed

Reply:
peek property is essentially doing the same.